devicetypes/smartthings/fidure-2nd-generation-thermostat.src/fidure-2nd-generation-thermostat.groovy

```diff
@@ -0,0 +1,810 @@
+			state "full", label:'Locked', action:"unlock", icon:"st.locks.lock.locked", backgroundColor:"#79b821", nextState:"unlocking"
+			state "level4", label:'Level 4', action:"unlock", icon:"st.locks.lock.locked", backgroundColor:"#79b821", nextState:"unlocking"
+			state "level3", label:'Level 3', action:"unlock", icon:"st.locks.lock.locked", backgroundColor:"#79b821", nextState:"unlocking"
+			state "setpoint", label:'Setpoint', action:"unlock", icon:"st.locks.lock.locked", backgroundColor:"#79b821", nextState:"unlocking"
+			state "modeonly", label:'Mode Only', action:"unlock", icon:"st.locks.lock.locked", backgroundColor:"#79b821", nextState:"unlocking"
+			state "unlocked", label:'Unlocked', action:"lock", icon:"st.locks.lock.unlocked", backgroundColor:"#ffffff", nextState:"locking"
+			state "locking", label:'Locking', icon:"st.locks.lock.locked", backgroundColor:"#79b821"
+			state "unlocking", label:'Unlocking', icon:"st.locks.lock.unlocked", backgroundColor:"#ffffff"
+			
+		}
+		
+		standardTile("refresh", "device.temperature", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "default", action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+
+		standardTile("hold", "setpointHold", width: 2, height: 2, decoration: "flat") {
+			state "changing", label:'Updating'
+			state "On", label:'Hold ${currentValue}', action:"turnHoldOff", nextState:"changing"
+			state "Off", label:'Hold ${currentValue}', action:"turnHoldOn", nextState:"changing"
+		}
+
+		standardTile("program", "prorgammingOperation", width: 2, height: 2, decoration: "flat") {
+			state "changing", label:'Updating'
+			state "On", label:'Sched ${currentValue}', action:"turnProgOff", nextState:"changing"
+			state "Off", label:'Sched ${currentValue}', action:"turnProgOn", nextState:"changing"
+		}
+
+		valueTile("humidity", "device.humidity", width: 2, height: 2, decoration: "flat") {
+			state "humidity", label:'${currentValue}%'
+		}
+		valueTile("seen", "lastSeen", width: 2, height: 2, decoration: "flat") {
+			state "seen", label:'last seen ${currentValue}'
+		}
+
+		main("thermostatFull")
+		details([
+			"thermostatFull", 
+			"temperature","tempDown","tempUp",
+			"mode", "fanMode", "operatingState",
+			"heatingSetpoint", "heatDown", "heatUp",
+			"coolingSetpoint", "coolDown", "coolUp",
+			"toggle","program", "hold",
+			"humidity","refresh", "seen"
+
+		])
+	}
+	
+}
+
+def installed() {
+	log.debug "installed"
+	// checkInterval of 12 minutes.
+    sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+
+	sendEvent(name: "temperature", value: 72, unit: "F")
+	sendEvent(name: "heatingSetpoint", value: 70, unit: "F")
+	sendEvent(name: "thermostatSetpoint", value: 70, unit: "F")
+	sendEvent(name: "coolingSetpoint", value: 76, unit: "F")
+	sendEvent(name: "thermostatMode", value: "off")
+	sendEvent(name: "thermostatFanMode", value: "fanAuto")
```

Comment:
thermostatMode and thermostatFanMode should have a data field specifying the supported modes, e.g.
data: [supportedThermostatModes: ["off", "heat", "cool"]]
data:[supportedThermostatFanModes: ["on", "auto"]]
Events for these attributes must also be sent, e.g
sendEvent(name: "supportedThermostatModes", value: ["off", "heat", "cool"], displayed: false)
sendEvent(name: "supportedThermostatFanModes", value: ["on", "auto"], displayed: false)
The values should preferable be obtained from the thermostat itself and not be hardcoded values.